### PR TITLE
Changes for #1021 to support targetAll translator property.

### PR DESCRIPTION
**Pull request by adomasven**

### CODE REVIEW

File: test/tests/server_connectorTest.js [hunk 1]
```diff
@@ -285,4 +285,29 @@ describe("Connector Server", function () {
 			);
 		});
 	});
+	
+	describe('/connector/getTranslatorCode', function() {
+		it('should respond with translator code', function* () {
+			var code = 'function detectWeb() {}\nfunction doImport() {}';
+			var translator = buildDummyTranslator(4, code);
+			sinon.stub(Zotero.Translators, 'get').returns(translator);
```
Comment (dstillman) [inline test/tests/server_connectorTest.js]: We should test that `Zotero.Translators.get` is getting a request for the right translator.

File: test/tests/server_connectorTest.js [hunk 2]
```diff
@@ -285,4 +285,29 @@ describe("Connector Server", function () {
...
+						translatorID: "dummy-translator",
+					})
+				}
+			);
+			
+			assert.equal(response.response, code);
+			
+			Zotero.Translators.get.restore();
+		})
+	});
```
Comment (dstillman) [inline test/tests/server_connectorTest.js]: Might as well move this above the other two `describe` blocks to be both alphabetically and chronologically correct.

File: chrome/content/zotero/xpcom/translation/translators.js [hunk 3]
```diff
@@ -46,7 +46,7 @@ Zotero.Translators = new function() {
 		Zotero.debug("Initializing translators");
 		var start = new Date;
 		
-		_cache = {"import":[], "export":[], "web":[], "search":[]};
+		_cache = {"import":[], "export":[], "web":[], "webWithTargetAll":[], "search":[]};
```
Comment (adomasven) [inline chrome/content/zotero/xpcom/translation/translators.js]: Suggestions for a better way to do this are welcome.

File: chrome/content/zotero/xpcom/translation/translators.js [hunk 4]
```diff
@@ -267,76 +270,92 @@ Zotero.Translators = new function() {
...
-						
-						if(j === 0) {
-							if(knownProxy) {
-								converterFunctions.push(Zotero.Proxies.properToProxy);
-							} else {
-								converterFunctions.push(null);
+			for (let translator of allTranslators) {
+				translatorLoop:
+				for (let rootSearchURI in rootSearchURIs) {
+					let isGeneric = (!translator.webRegexp.root && translator.runMode === Zotero.Translator.RUN_MODE_IN_BROWSER);
```
Comment (dstillman) [inline chrome/content/zotero/xpcom/translation/translators.js]: I don't think `RUN_MODE_IN_BROWSER` is related to generic, so I don't think it should be combined (which also makes the below condition rather confusing). Can use a different variable for the other checks.

**Pull request merged**